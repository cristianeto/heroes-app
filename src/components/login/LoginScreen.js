import React, { useContext } from "react";
import { types } from "../../types/types";
import { AuthContext } from "../../auth/AuthContext";

const LoginScreen = ({ history }) => {
  const { dispatch } = useContext(AuthContext);

  const handleClick = () => {
    //? Diferencia entre psuh y replace...
    //? replace no guarda la historia de navegacion, no permite ir ATRAS
    //history.push("/");

    const action = {
      type: types.login,
      payload: { name: "Cristian" },
    };
    dispatch(action);
    history.replace("/");
  };
  return (
    <div className="container mt-5">
      <h1>LoginScreen</h1>
      <hr />
      <button className="btn btn-primary" onClick={handleClick}>
        Login
      </button>
    </div>
  );
};

export default LoginScreen;
