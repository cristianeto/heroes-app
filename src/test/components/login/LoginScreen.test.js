import { mount } from 'enzyme';
import LoginScreen from '../../../components/login/LoginScreen';
import { AuthContext } from '../../../auth/AuthContext';
import { types } from '../../../types/types';

describe('Testing in <LoginScreen /> component', () => {
  const historyMock = {
    replace: jest.fn(),
  };

  const contextValue = {
    user: { logged: false },
    dispatch: jest.fn(),
  };

  const wrapper = mount(
    <AuthContext.Provider value={contextValue}>
      <LoginScreen history={historyMock} />
    </AuthContext.Provider>
  );

  test('should show the componentc correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should call dispatch method and call replace method of history', () => {
    const action = {
      payload: { name: 'Cristian' },
      type: types.login,
    };
    wrapper.find('button').prop('onClick')();
    expect(contextValue.dispatch).toHaveBeenCalledWith(action);
    expect(historyMock.replace).toHaveBeenCalledTimes(1);
    expect(historyMock.replace).toHaveBeenCalledWith('/');
  });
});
