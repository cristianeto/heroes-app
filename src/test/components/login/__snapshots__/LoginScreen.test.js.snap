// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in <LoginScreen /> component should show the componentc correctly 1`] = `
<div
  className="container mt-5"
>
  <h1>
    LoginScreen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
